src/request.ts

```diff
@@ -0,0 +1,37 @@
+import { ErrorResponse, NetworkError, Headers } from './base-client';
+import { Logger } from './logger';
+import { RetryStrategy } from './retry-strategy/retry-strategy';
+
+//Single request
+export type NetworkRequest<T> = (parameters?: any) => Promise<T>;
+
+export interface RequestOptions {
+    method: string;
+    path: string;
+    jwt?: string;
+    headers?: Headers;
+    body?: any;
+    retryStrategy?: RetryStrategy;
+    logger?: Logger;
+}
+
+export function executeRequest<T>(createXhr: () => XMLHttpRequest, options: RequestOptions): Promise<any> {
```

Comment:
Was the return type supposed to be `Promise<T>` or does this function not need to be templated?